bin/bridge-client/src/modes/operator/task_manager.rs

```diff
@@ -54,22 +64,26 @@ where
                 });
             }
 
-            let any_failed = handles.join_all().await.iter().any(|res| res.is_err());
-
-            // if none of the duties failed, update the duty index so that the
-            // next batch is fetched in the next poll.
-            //
-            // otherwise, don't update the index so that the current batch is refetched and
-            // ones that were not executed successfully are executed again.
-            if !any_failed {
-                info!(%start_index, %stop_index, "updating duty index");
-                if let Err(e) = self
-                    .bridge_duty_idx_db_ops
-                    .set_index_async(stop_index)
-                    .await
-                {
-                    error!(error = %e, %start_index, %stop_index, "could not update duty index");
+            // TODO: There should be timeout duration based on duty and not a common timeout
+            // duration
+            if let Ok(any_failed) = timeout(duty_timeout_duration, handles.join_all()).await {
+                // if none of the duties failed, update the duty index so that the
+                // next batch is fetched in the next poll.
+                //
+                // otherwise, don't update the index so that the current batch is refetched and
+                // ones that were not executed successfully are executed again.
+                if !any_failed.iter().any(|res| res.is_err()) {
+                    info!(%start_index, %stop_index, "updating duty index");
+                    if let Err(e) = self
+                        .bridge_duty_idx_db_ops
+                        .set_index_async(stop_index)
+                        .await
+                    {
+                        error!(error = %e, %start_index, %stop_index, "could not update duty index");
+                    }
                 }
+            } else {
+                error!(?duty_timeout_duration, "some duties timed out");
```

Comment:
Can use % here, although I'm not sure how useful it is to even have this here.

Reply:
Duration doesn't have default formatter. But i think removing this makes sense as we don't exactly track which duty got timed out